src/components/Reactions/Reactions.tsx

```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+
+import {FaceSmile} from '@gravity-ui/icons';
+import {
+    Button,
+    DOMProps,
+    Flex,
+    Icon,
+    Palette,
+    PaletteProps,
+    Popover,
+    QAProps,
+} from '@gravity-ui/uikit';
+import xor from 'lodash/xor';
+
+import {block} from '../utils/cn';
+import {useStableCallback} from '../utils/useStableCallback';
+
+import {Reaction, ReactionProps} from './Reaction';
+import {ReactionsContextProvider, ReactionsContextTooltipProps} from './context';
+
+import './Reactions.scss';
+
+const b = block('reactions');
+
+export type ReactionsPaletteProps = Omit<
+    PaletteProps,
+    'value' | 'defaultValue' | 'onUpdate' | 'size' | 'disabled' | 'multiple'
+>;
+
+export interface ReactionsProps extends Pick<PaletteProps, 'size' | 'disabled'>, QAProps, DOMProps {
```

Comment:
Why we need `disabled`? What is use case for that?

Reply:
Now disabled buttons look barely visible, I think it's a bad UX

Reply:
Do we have a button state for that?
Because simply disabling the `onClickReaction` callback looks like a bad UX as well: a user hovers a button, it reacts (the background changes), he clicks on it, the button reacts again (it changes it's size slightly and goes back) and then simply nothing happens — that would be quite confusing